drivers/sensor/bmi270/bmi270.c

```diff
@@ -0,0 +1,714 @@
+
+		ret = reg_write(BMI270_REG_INIT_ADDR_0, addr_array, 2, dev);
+
+		k_usleep(450);
+
+		if (ret == 0) {
+			ret = reg_write(BMI270_REG_INIT_DATA,
+					(bmi270_config_file + index),
+					BMI270_WR_LEN, dev);
+			k_usleep(450);
+		}
+	}
+
+	return ret;
+}
+
+int bmi270_init(const struct device *dev)
+{
+	int ret;
+	struct bmi270_data *drv_dev = dev->data;
+	const struct bmi270_dev_config *cfg = dev->config;
+	uint8_t chip_id;
+	uint8_t soft_reset_cmd;
+	uint8_t init_ctrl;
+	uint8_t msg;
+	uint8_t tries;
+	uint8_t adv_pwr_save;
+
+	drv_dev->i2c = device_get_binding(cfg->i2c_master_name);
+	if (drv_dev->i2c == NULL) {
+		LOG_ERR("Could not get pointer to %s device",
+			cfg->i2c_master_name);
+		return -EINVAL;
+	}
+
+	drv_dev->i2c_addr = cfg->i2c_addr;
+	drv_dev->acc_odr = BMI270_ACC_ODR_100_HZ;
+	drv_dev->acc_range = 8;
+	drv_dev->gyr_odr = BMI270_GYR_ODR_200_HZ;
+	drv_dev->gyr_range = 2000;
+
+	k_usleep(BMI270_POWER_ON_TIME);
+
+	ret = reg_read(BMI270_REG_CHIP_ID, &chip_id, 1, drv_dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	if (chip_id != BMI270_CHIP_ID) {
+		LOG_ERR("Unexpected chip id (%x). Expected (%x)",
+			chip_id, BMI270_CHIP_ID);
+		return -EIO;
+	}
+
+	soft_reset_cmd = BMI270_CMD_SOFT_RESET;
+	ret = reg_write(BMI270_REG_CMD, &soft_reset_cmd, 1, drv_dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	k_usleep(BMI270_SOFT_RESET_TIME);
+
+	ret = reg_read(BMI270_REG_PWR_CONF, &adv_pwr_save, 1, drv_dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	adv_pwr_save = BMI270_SET_BITS_POS_0(adv_pwr_save,
+					     BMI270_PWR_CONF_ADV_PWR_SAVE,
+					     BMI270_PWR_CONF_ADV_PWR_SAVE_DIS);
+	ret = reg_write(BMI270_REG_PWR_CONF, &adv_pwr_save, 1, drv_dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	init_ctrl = BMI270_PREPARE_CONFIG_LOAD;
+	ret = reg_write(BMI270_REG_INIT_CTRL, &init_ctrl, 1, drv_dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = write_config_file(drv_dev);
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	init_ctrl = BMI270_COMPLETE_CONFIG_LOAD;
+	ret = reg_write(BMI270_REG_INIT_CTRL, &init_ctrl, 1, drv_dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* Timeout after 150ms. If tries is 0 by the end of the loop,
+	 * report an error
+	 */
+	for (tries = 0; tries < 16; tries++) {
+		ret = reg_read(BMI270_REG_INTERNAL_STATUS, &msg, 1, drv_dev);
+		if (ret != 0) {
+			return ret;
+		}
+
+		msg &= BMI270_INST_MESSAGE_MSK;
+		if (msg == BMI270_INST_MESSAGE_INIT_OK) {
+			break;
+		}
+
+		k_usleep(10000);
+	}
+
+	if (tries == 0) {
+		return -EIO;
+	}
+
+	adv_pwr_save = BMI270_SET_BITS_POS_0(adv_pwr_save,
+					     BMI270_PWR_CONF_ADV_PWR_SAVE,
+					     BMI270_PWR_CONF_ADV_PWR_SAVE_EN);
+	ret = reg_write(BMI270_REG_PWR_CONF, &adv_pwr_save, 1, drv_dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	return ret;
+}
+
+int bmi270_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct bmi270_data *drv_dev = dev->data;
+	uint8_t data[12];
+	int ret;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
```

Comment:
I think error handling (e.g. `-ENOTSUP`) should be used here instead of an assertion

Reply:
As get supports handling individual/group channels, fetch should do the same.